nixos/modules/services/backup/restic.nix

```diff
@@ -376,5 +376,21 @@ in
           timerConfig = backup.timerConfig;
         })
         config.services.restic.backups;
+
+    environment.systemPackages = lib.mapAttrsToList (name: backup: let
+      extraOptions = lib.concatMapStrings (arg: " -o ${arg}") backup.extraOptions;
+      resticCmd = "${backup.package}/bin/restic${extraOptions}";
+    in pkgs.writeShellScriptBin "restic-${name}" ''
+      set -a  # automatically export variables
+      ${lib.optionalString (backup.environmentFile != null) "source ${backup.environmentFile}"}
+      ${lib.optionalString (backup.s3CredentialsFile != null) "source ${backup.s3CredentialsFile}"}
+      RESTIC_CACHE_DIR="/var/cache/restic-backups-${name}"
+      ${lib.concatStringsSep "\n" (
+        lib.mapAttrsToList (n: v: "${n}=${v}")
+        (lib.filterAttrs (_: v: v != null) config.systemd.services."restic-backups-${name}".environment)
+      )}
```

Comment:
this is setting all the environment variables from the systemd unit isn't it? Maybe add a comment saying so if that's the case since it took me quite a while parsing what this is doing.